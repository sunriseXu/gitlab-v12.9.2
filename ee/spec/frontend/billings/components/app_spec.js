import { shallowMount } from '@vue/test-utils';
import createStore from 'ee/billings/stores';
import SubscriptionApp from 'ee/billings/components/app.vue';
import SubscriptionTable from 'ee/billings/components/subscription_table.vue';

describe('SubscriptionApp component', () => {
  let store;
  let wrapper;

  const appProps = {
    namespaceId: '42',
    namespaceName: 'bronze',
    planUpgradeHref: '/url',
    customerPortalUrl: 'https://customers.gitlab.com/subscriptions',
  };

  const factory = (props = appProps) => {
    store = createStore();
    jest.spyOn(store, 'dispatch').mockImplementation();

    wrapper = shallowMount(SubscriptionApp, {
      store,
      propsData: { ...props },
    });
  };

  const expectComponentWithProps = (Component, props = {}) => {
    const componentWrapper = wrapper.find(Component);

    expect(componentWrapper.isVisible()).toBeTruthy();
    expect(componentWrapper.props()).toEqual(expect.objectContaining(props));
  };

  afterEach(() => {
    wrapper.destroy();
  });

  describe('on creation', () => {
    beforeEach(() => {
      factory();
    });

    it('dispatches the setNamespaceId on mounted', () => {
      expect(store.dispatch.mock.calls).toEqual([
        ['subscription/setNamespaceId', appProps.namespaceId],
      ]);
    });

    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('passes the correct props to the subscriptions table', () => {
      expectComponentWithProps(SubscriptionTable, {
        namespaceName: appProps.namespaceName,
        planUpgradeHref: appProps.planUpgradeHref,
        customerPortalUrl: appProps.customerPortalUrl,
      });
    });
  });
});
