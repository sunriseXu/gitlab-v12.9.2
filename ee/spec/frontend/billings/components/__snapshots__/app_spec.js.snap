// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscriptionApp component on creation matches the snapshot 1`] = `
<subscription-table-stub
  customerportalurl="https://customers.gitlab.com/subscriptions"
  namespacename="bronze"
  planupgradehref="/url"
/>
`;
